lib/xml-utils.coffee

```diff
@@ -51,10 +58,10 @@ module.exports =
 
       for match in matches ? []
         # Start comment
-        if match == "<!--"
+        if match == "<!--" || match == "<![CDATA["
```

Comment:
What will happen if there is a comment inside the CDATA block? I think that it would need a similar approach as for comments but with a new variable `waitingStartCDATA`, so it can handle both comments and CDATA block.